provider/redis.go

```diff
@@ -34,58 +38,95 @@ func redisOnlineStoreFactory(serialized pc.SerializedConfig) (Provider, error) {
 	if redisConfig.Prefix == "" {
 		redisConfig.Prefix = "Featureform_table__"
 	}
-	return NewRedisOnlineStore(redisConfig), nil
+	return NewRedisOnlineStore(redisConfig)
 }
 
-func NewRedisOnlineStore(options *pc.RedisConfig) *redisOnlineStore {
-	redisOptions := &redis.Options{
-		Addr:     options.Addr,
-		Password: options.Password,
-		DB:       options.DB,
+func NewRedisOnlineStore(options *pc.RedisConfig) (*redisOnlineStore, error) {
+	redisOptions := rueidis.ClientOption{
+		InitAddress: []string{options.Addr},
+		Password:    options.Password,
+		SelectDB:    options.DB,
+	}
+	redisClient, err := rueidis.NewClient(redisOptions)
+	if err != nil {
+		return nil, err
 	}
-	redisClient := redis.NewClient(redisOptions)
 	return &redisOnlineStore{redisClient, options.Prefix, BaseProvider{
 		ProviderType:   pt.RedisOnline,
 		ProviderConfig: options.Serialized(),
 	},
-	}
+	}, nil
 }
 
 func (store *redisOnlineStore) AsOnlineStore() (OnlineStore, error) {
 	return store, nil
 }
 
+func (store *redisOnlineStore) AsVectorStore() (VectorStore, error) {
+	return store, nil
+}
+
 func (store *redisOnlineStore) Close() error {
-	// from the docs:
-	//
-	// "It is rare to Close a Client, as the Client is meant to be
-	// long-lived and shared between many goroutines.""
-	//
-	return store.client.Close()
+	store.client.Close()
```

Comment:
Does this .Close() not return anything?

Reply:
With the new Redis client, the `Close` method returns nothing, oddly.